### PR TITLE
Rename some return types and options

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 1]
```diff
@@ -10,21 +10,21 @@
...
 - Moved `SecondaryHostForRetryReads` out of retry options, now it's under `BlobClientOptions`.
 - Changed the return types of the following APIs:
-  - `BlobServiceClient::DeleteBlobContainer` now returns `Azure::Core::Response<Models::DeleteBlobContainerResult>`.
+  - `BlobServiceClient::DeleteBlobContainer` now returns `DeleteBlobContainerResult`.
   - `BlobServiceClient::GetUserDelegationKey` now returns `UserDelegationKey`.
   - `BlobServiceClient::GetProperties` now returns `BlobServiceProperties`.
   - `BlobServiceClient::GetAccountInfo` now returns `AccountInfo`.
   - `BlobServiceClient::GetStatistics` now returns `ServiceStatistics`.
-  - `BlobContainerClient::DeleteBlob` now returns `Azure::Core::Response<Models::DeleteBlobResult>`.
+  - `BlobContainerClient::DeleteBlob` now returns `Models::DeleteBlobResult`.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:22]:
```suggestion
  - `BlobContainerClient::DeleteBlob` now returns `DeleteBlobResult`.
```

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 2]
```diff
@@ -10,21 +10,21 @@
...
   - `BlobContainerClient::GetProperties` now returns `BlobContainerProperties`.
   - `BlobContainerClient::GetAccessPolicy` now returns `BlobContainerAccessPolicy`.
   - `BlobClient::GetProperties` now returns `BlobProperties`.
   - `BlobClinet::GetTags` now returns `std::map<std::string, std::string>`.
-- Removed `PreviousShareSnapshot` from `GetShareFileRangeListOptions`, use `ShareFileClient::GetRangeListDiff` instead.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:27]: Why remove this? Was this intentional? If not, add it back.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp [hunk 3]
```diff
@@ -947,7 +947,7 @@ namespace Azure { namespace Storage { namespace Blobs {
   /**
    * @brief Optional parameters for PageBlobClient::UploadPages.
    */
-  struct UploadPageBlobPagesOptions
+  struct UploadPagesOptions
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp:950]: Add CL entry for dropping `PageBlob`

File: sdk/storage/azure-storage-blobs/CHANGELOG.md (hunk 2 above)
Comment (ahsonkhan) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:27]: Oh I get it, it was in the wrong package. Good catch.

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp [hunk 4]
```diff
@@ -133,8 +133,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
       DownloadFileDetails Details;
     };
 
-    using StartCopyShareFileResult = _detail::FileStartCopyResult;
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp:136]: Was this not used anywhere? Add to CL?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp:136]: > Changed the return type of StartCopy API from a Response<T> to the particular Operation<T> type called StartCopyShareFileOperation directly.

This entry should have covered that.

**Review by katmsft — APPROVED** [review]

**Pull request merged**